iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -74,7 +75,7 @@ public async Task RunSampleAsync(TimeSpan sampleRunningTime)
             try
             {
                 await InitializeAndSetupClientAsync(s_cancellationTokenSource.Token);
-                await Task.WhenAll(SendMessagesAsync(s_cancellationTokenSource.Token), ReceiveMessagesAsync(s_cancellationTokenSource.Token));
+                await Task.WhenAll(GetTwinUpdateAsync(s_cancellationTokenSource.Token), SendMessagesAsync(s_cancellationTokenSource.Token), ReceiveMessagesAsync(s_cancellationTokenSource.Token));
```

Comment:
We wouldn't want to call "GetTwin()" continuously on a loop, but rather only when the client is (re)connected. You can merge the logic you have in here with the "GetTwin" logic that you have in "InitializeAndSetupClientAsync".
On client (re)connect:
get twin -> compare the desired property version with last retrieved version -> if local version is less than server version - perform property update -> end

Reply:
The subsequent desired property updates would be handled by the "SetDesiredPropertyUpdateCallbackAsync" that is set in "InitializeAndSetupClientAsync".

Reply:
From my tests, in the case that connection status changes from **Disconnected_Retrying** to **Connected**, twin updates on the server side during disconnection period  will not be handled by `SetDesiredPropertyUpdateCallbackAsync()` after the client gets re-connceted. That's why I wanted to call "GetTwin()" continuously on a loop. 

Meanwhile I am checking with Python and Node SDK teams as well to understand their behaviours.